rotkehlchen/chain/evm/decoding/cowswap/decoder.py

```diff
@@ -369,7 +372,27 @@ def _decode_normal_claim(self, context: DecoderContext) -> DecodingOutput:
         raw_amount = hex_or_bytes_to_int(context.tx_log.data[128:160])
         amount = asset_normalized_value(amount=raw_amount, asset=self.vcow_token)
         airdrop_identifier: Literal['cow_mainnet', 'cow_gnosis'] = 'cow_mainnet' if self.evm_inquirer.chain_id == ChainID.ETHEREUM else 'cow_gnosis'  # noqa: E501
+        # claimTypes with payment: 1=GnoOption, 2=UserOption, 3=Investor
+        claim_supports_payment = hex_or_bytes_to_int(context.tx_log.data[32:64]) in (1, 2, 3)
+        claimant_address = hex_or_bytes_to_address(context.tx_log.data[64:96])
+        claim_has_payment = False
+        out_event = in_event = None
         for event in context.decoded_events:
+            # Claim event always follows payment. Continue on payment, break on claim.
+            if (
+                claim_supports_payment and
+                self.base.is_tracked(claimant_address) and
```

Comment:
claimant_address doesn't get modified so you don't need to check this on every iteration of the loop. If it is not tracked you need to exit this logic but before entering the for loop